microsoft/iot-cardboard-js
# [Data History Explorer]: Property Inspector Integration

### Summary of changes 🔍 
- Added `DataHistoryExplorerModalControl` component to wrap the modal with controlled chart icon
- Integrated it into the higher component of PropertyInspector component (where it is actually triggered is in the **PropertyTree** of the StandalonePropertyInspector to align it with the $dtId of the selected twin in the DOM)
- Currently only showing the data history icon in the Viewer's **BehaviorModal**'s PropertyInspector and _only active when the feature flag is on_
- Added some stories to show the use case of the data history enabled feature property inspector
- Made some type/interface changes
   - `ADT3DScenePageAdapter` is now extending `ADTDataHistoryAdapter` instead of extending AzureManagementAdapter, ADTAdapter and ADXAdapter individually since ADTDataHistoryAdapter is already a mixin of those.
- Changed twin id in mock data
- Exported some components to consume in the PublicRepoFork

**Merge https://github.com/microsoft/iot-cardboard-js/pull/842 first to get less file diffs before reviewing.**


![image](https://user-images.githubusercontent.com/45217314/216166179-8d5caf44-b4f2-430e-ad7f-7280e606cf54.png)
![image](https://user-images.githubusercontent.com/45217314/216166259-44260299-09bd-4949-b35a-62fb9fefb967.png)
![image](https://user-images.githubusercontent.com/45217314/216166071-4adc1837-5723-4a5e-8c7a-c65f92712501.png)
![image](https://user-images.githubusercontent.com/45217314/216166390-a28a7b9d-e24c-42b8-bce4-26ce6768cd4d.png)

<img width="482" alt="image" src="https://user-images.githubusercontent.com/45217314/216165716-c9695f8e-54f6-4aaf-af5a-a0059a6771e8.png">

![image](https://user-images.githubusercontent.com/45217314/217690021-3093126c-b16f-413e-a31b-f4e27a5264b6.png)
![image](https://user-images.githubusercontent.com/45217314/217690406-98b9e0fb-0bb5-44a1-ae17-d4fdacad1de7.png)

<img width="964" alt="image" src="https://user-images.githubusercontent.com/45217314/216169892-c7713b55-274b-4d93-8a84-feb75b87999e.png">

![image](https://user-images.githubusercontent.com/45217314/216170776-8e8d0b0a-72e0-42c4-ba9e-cb523688e6c8.png)
![image](https://user-images.githubusercontent.com/45217314/216170502-25f5d2f5-2980-4ab2-b77c-51d58fff1403.png)

### Testing 🧪
- localStorage.setItem('cardboard.feature.dataHistoryExplorer',true)
- Check it on WithDataHistoryIconEnabled or WithDataHistoryIconDisabled PropertyInspector mock stories or AdtTwinWithDataHistory local story OR,
- Check out ADT3DScenePage stories with MockAdapter and ADT and go to "View" after selecting a scene and click on one of the alerts to see the twin behavior and select "All properties" tab to see the PropertyInspector. In local story if your instance is supporting data history you will see the icon enabled. Otherwise it will be disabled.

### Checklist ✔️
